aiida_common_workflows/workflows/relax/gpaw/workchain.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+"""Implementation of `aiida_common_workflows.common.relax.workchain.CommonRelaxWorkChain` for GPAW."""
+from aiida import orm
+from aiida.engine import calcfunction
+from aiida.plugins import WorkflowFactory
+
+from ..workchain import CommonRelaxWorkChain
+from .generator import GPAWCommonRelaxInputGenerator
+
+__all__ = ('GPAWCommonRelaxWorkChain',)
+
+
+@calcfunction
+def extract_from_array(array):
+    """Return the forces and stress arrays from the given trajectory data."""
+    forces = orm.ArrayData()
+    forces.set_array(name='forces', array=array.get_array('forces'))
+
+    return {'forces': forces}
+
+
+@calcfunction
+def extract_from_parameters(parameters):
+    """Return the total energy from the given parameters node."""
+    energy_cont = parameters.get_attribute('energy_contributions')
+    total_energy =  energy_cont['xc'] + energy_cont['local'] + energy_cont['kinetic']
+    total_energy += energy_cont['external'] + energy_cont['potential'] + energy_cont['entropy (-st)']
+    results = {'total_energy': orm.Float(total_energy)}
+
+    return results
+
+
+class GPAWCommonRelaxWorkChain(CommonRelaxWorkChain):
+    """Implementation of `aiida_common_workflows.common.relax.workchain.CommonRelaxWorkChain` for GPAW."""
+
+    _process_class = WorkflowFactory('ase.gpaw.base')
+    _generator_class = GPAWCommonRelaxInputGenerator
+
+    def convert_outputs(self):
+        """Convert the outputs of the sub workchain to the common output specification."""
+        outputs = self.ctx.workchain.outputs
+
+        total_energy, = extract_from_parameters(outputs.parameters).values()
+        forces, = extract_from_array(outputs.array).values()
+
+        if 'output_structure' in outputs:
+            self.out('relaxed_structure', outputs.output_structure)
+
+        self.out('total_energy', total_energy)
+        self.out('forces', forces)
```

Comment:
I am not sure this works. Have you tested to run this? the calcfunctions return a dictionary, so `total_energy` and `forces` variables should be dicts as well. So either you change the calcfunction to just simply return the `Data` output node (not a dictionary, which is not required for a single output) or, you dereference:
```
self.out('total_energy', total_energy['total_energy'])
```
I would suggest the former

Reply:
Not sure why I had written it this way, but there is a `.values()`  after the function call to `extract_from_parameters`  (function names have now changed); with the `,` after `total_energies` and `forces` these are actually floats or lists and not dicts. 

That being said, I have now updated this function with your first suggestion.